recipes/coin-cbc/all/conanfile.py

```diff
@@ -34,32 +32,44 @@ def _source_subfolder(self):
     def _build_subfolder(self):
         return "build_subfolder"
 
+    def export_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
-        if self.settings.os == "Windows" and self.options.shared:
-            raise ConanInvalidConfiguration("coin-cbc does not support shared builds on Windows")
         if self.options.shared:
             del self.options.fPIC
 
+    def validate(self):
+        if self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("coin-cbc does not support shared builds on Windows")
+        if hasattr(self, "settings_build") and tools.cross_building(self):
+            raise ConanInvalidConfiguration("Cross building is not yet supported. Contributions are welcome")
```

Comment:
Did you try the tricks from https://github.com/conan-io/conan-center-index/pull/7162?

Reply:
I did not, because it was not my goal to add cross-compilation support to the recipe.
Is the `trick` to use _user_info_build instead of deps_user_info ? Or is also gnu-config part of the trick ?

Reply:
gnu-config is the trick.